### PR TITLE
add renovate to update exporter versions

### PR DESCRIPTION
**Pull request by TheMeier**
This ads a renovate github action which will create pull requests for updated versions of components.
Currently 16 (of 41) versions are set in `data/defaults.yaml`, more to follow soon

### CODE REVIEW

File: data/defaults.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ prometheus::pushprox_client::group: 'pushprox_client'
 prometheus::pushprox_client::package_ensure: 'latest'
 prometheus::pushprox_client::package_name: 'pushprox_client'
 prometheus::pushprox_client::user: 'pushprox_client'
+# renovate: depName=prometheus-community/PushProx
```
Comment (bastelfreak) [inline data/defaults.yaml:9]: is there a chance that renovate can parse the puppet code? we moved most of the data out of defaults.yaml into the puppet classes so puppet-strings can pick up the data. We just didn't complete this migration yet.

Comment (TheMeier) [reply data/defaults.yaml:9]: Well the whole thing is based on regex. So if we can harmonize all modules to use the same format, eg:
```
 String[1] $version                                         = 'x.x.x',
```
that is should be doable. In my mind it was much nicer to have the versions centralized in the module-data. Do we need the versions in the `REFERENCE.md`?

Comment (TheMeier) [reply data/defaults.yaml:9]: I have added support for the versions in the pp files

**Comment by bastelfreak** [comment]
In general I like the idea. A few thoughts:
* Can you document this in the README.md or so? Because we need to ensure that we add the comments for new exporters as well
* How do we want to treat the the PRs renovate creates? In the past this was often a backwards-incompatible change, but one could also argue it's an enhancement if the upgrade from the old to new version works without issues
* Do you know if we can use dependabot for actions/checkout@v4.1.6 / renovatebot/github-action@v40.1.11? Or don't we want that? We don't run dependabot at the moment in our modules because we manage the CI configs via modulesync.

**Comment by TheMeier** [comment]
> In general I like the idea. A few thoughts:
> 
> * Can you document this in the README.md or so? Because we need to ensure that we add the comments for new exporters as well

Will do

> * How do we want to treat the the PRs renovate creates? In the past this was often a backwards-incompatible change, but one could also argue it's an enhancement if the upgrade from the old to new version works without issues

Hm I would argue that version updates are only enhancements in the majority of cases. But I guess there is no way around the reviewer to actually classify each PR desperately

> * Do you know if we can use dependabot for actions/checkout@v4.1.6 / renovatebot/github-action@v40.1.11? Or don't we want that? We don't run dependabot at the moment in our modules because we manage the CI configs via modulesync.

I only had a brief look into dependabot. I think this is possible. Something like this might do the trick:
```yaml
  - package-ecosystem: "github-actions"
    directory: "/"
    schedule:
      interval: "weekly"
      day: "sunday"
      time: "16:00"
    groups:
      renovate:
        patterns: [ "actions/checkout" ,"renovatebot/github-action]
```

**Comment by TheMeier** [comment]
I have added a dependabot config that **only** updates `renovatebot/github-action` and ignores the others. Sadly you have to list the ignores explicitly, meaning if more actions will be added some day the ignore list needs to be updated.
There is no way to limit dependabot to specific files also sub-directories for workflows are not supported, so  using https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#directory is also not an option.

This is the log from my tests: https://github.com/TheMeier/puppet-prometheus/network/updates/836753113

**Comment by TheMeier** [comment]
We could also allow actions/checkout version "4.x" but that will catch other workflows once the checkout action gets updated there by modulesync some day

**Comment by TheMeier** [comment]
Hm it seems in order for renovate to work correctly a custom personal access token i needed after all:

https://github.com/renovatebot/github-action?tab=readme-ov-file#token
> In particular, using the GITHUB_TOKEN to create a new Pull Request from more types of Github Workflows results in Pull Requests that do not trigger your Pull Request and Push CI events.

But I am not sure if that really is necessary for this use-case

**Review by bastelfreak — APPROVED** [review]

**Review by tuxmea — APPROVED** [review]

**Comment by TheMeier** [comment]
Well this did not work out very well. The token permissions are not sufficient, also the created PRs do not run the tests, as documented in https://github.com/renovatebot/github-action?tab=readme-ov-file#token, even though `workflow_dispatch` is set. 
In my tests with a fork both of these aspects did work. Unless we have a way to configure a personal access token (possibly from some tool-user) I don't see any way this works.

**Comment by TheMeier** [comment]
Another issue with this REFERENCE.md needs to be recreated in this workflow.

**Pull request merged**